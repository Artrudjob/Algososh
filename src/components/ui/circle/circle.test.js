import React from "react";
import {create} from "react-test-renderer";
import {Circle} from "./circle";
import {ElementStates} from "../../../types/element-states";

describe("TEST CIRCLE", () => {
    test("Circle with not a letter", () => {
        const componentCircle = create(<Circle />);
        expect(componentCircle.toJSON()).toMatchSnapshot();
    })

    test("Circle with a letter", () => {
        const componentCircle = create(<Circle letter={"test"} />);
        expect(componentCircle.toJSON()).toMatchSnapshot();
    })

    test("Circle with head", () => {
        const componentCircle = create(<Circle head={"test"} />);
        expect(componentCircle.toJSON()).toMatchSnapshot();
    })

    test("Circle with a JSX.element in the head", () => {
        const componentCircle = create(<Circle head={<Circle />} />);
        expect(componentCircle.toJSON()).toMatchSnapshot();
    })

    test("Circle with tail", () => {
        const componentCircle = create(<Circle tail={"test"} />);
        expect(componentCircle.toJSON()).toMatchSnapshot();
    })

    test("Circle with a JSX.element in the tail", () => {
        const componentCircle = create(<Circle tail={<Circle />} />);
        expect(componentCircle.toJSON()).toMatchSnapshot();
    })

    test("Circle with index", () => {
        const componentCircle = create(<Circle index={0} />);
        expect(componentCircle.toJSON()).toMatchSnapshot();
    })

    test("Circle with prop isSmall === true", () => {
        const componentCircle = create(<Circle isSmall={true} />);
        expect(componentCircle.toJSON()).toMatchSnapshot();
    })

    test("Circle with state === ElementStates.Default", () => {
        const componentCircle = create(<Circle state={ElementStates.Default} />);
        expect(componentCircle.toJSON()).toMatchSnapshot();
    })

    test("Circle with state === ElementStates.Changing", () => {
        const componentCircle = create(<Circle state={ElementStates.Changing} />);
        expect(componentCircle.toJSON()).toMatchSnapshot();
    })

    test("Circle with state === ElementStates.Modified", () => {
        const componentCircle = create(<Circle state={ElementStates.Modified} />);
        expect(componentCircle.toJSON()).toMatchSnapshot();
    })
})