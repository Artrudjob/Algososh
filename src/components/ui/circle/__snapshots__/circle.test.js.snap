// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST CIRCLE Circle with a JSX.element in the head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="circleHead"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-cy="circleHead"
      />
      <div
        className="circle   default"
        data-cy="circle"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-cy="circleTail"
      />
    </div>
  </div>
  <div
    className="circle   default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="circleTail"
  />
</div>
`;

exports[`TEST CIRCLE Circle with a JSX.element in the tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="circleHead"
  />
  <div
    className="circle   default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="circleTail"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-cy="circleHead"
      />
      <div
        className="circle   default"
        data-cy="circle"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-cy="circleTail"
      />
    </div>
  </div>
</div>
`;

exports[`TEST CIRCLE Circle with a letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="circleHead"
  />
  <div
    className="circle   default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      test
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="circleTail"
  />
</div>
`;

exports[`TEST CIRCLE Circle with head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
    data-cy="circleHead"
  >
    test
  </div>
  <div
    className="circle   default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="circleTail"
  />
</div>
`;

exports[`TEST CIRCLE Circle with index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="circleHead"
  />
  <div
    className="circle   default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
    data-cy="circleTail"
  />
</div>
`;

exports[`TEST CIRCLE Circle with not a letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="circleHead"
  />
  <div
    className="circle   default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="circleTail"
  />
</div>
`;

exports[`TEST CIRCLE Circle with prop isSmall === true 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="circleHead"
  />
  <div
    className="circle  small default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="circleTail"
  />
</div>
`;

exports[`TEST CIRCLE Circle with state === ElementStates.Changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="circleHead"
  />
  <div
    className="circle   changing"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="circleTail"
  />
</div>
`;

exports[`TEST CIRCLE Circle with state === ElementStates.Default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="circleHead"
  />
  <div
    className="circle   default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="circleTail"
  />
</div>
`;

exports[`TEST CIRCLE Circle with state === ElementStates.Modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="circleHead"
  />
  <div
    className="circle   modified"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="circleTail"
  />
</div>
`;

exports[`TEST CIRCLE Circle with tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="circleHead"
  />
  <div
    className="circle   default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
    data-cy="circleTail"
  >
    test
  </div>
</div>
`;
